File: yjKim513/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> players;
    
    for (auto iter = participant.begin(); iter != participant.end(); iter++) {
        if (players.find(*iter) == players.end()) {
            players.insert({*iter, 1});
        }
        else {
            (players.find(*iter)->second)++;
        }
    }
    
    for (auto iter = completion.begin(); iter != completion.end(); iter++) {
        if (--(players.find(*iter)->second) == 0)
            players.erase(*iter);
    }
    
    return players.begin()->first;
}

File: yjKim513/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    players = defaultdict(int)
    
    for player in participant:
        players[player] += 1
    for player in completion:
        players[player] -= 1
        if players[player] == 0:
            del players[player]
            
    return list(players.keys())[0]
    